packages/e2e-test-utils-playwright/src/page/press-key-with-modifier.ts

```diff
@@ -0,0 +1,138 @@
+/**
+ * External dependencies
+ */
+import { capitalize } from 'lodash';
+import type { Page } from '@playwright/test';
+
+/**
+ * WordPress dependencies
+ */
+import { modifiers, SHIFT, ALT, CTRL } from '@wordpress/keycodes';
+import type { WPKeycodeModifier } from '@wordpress/keycodes';
+
+/**
+ * Internal dependencies
+ */
+import type { PageUtils } from './index';
+
+let clipboardDataHolder: {
+	plainText: string;
+	html: string;
+} = {
+	plainText: '',
+	html: '',
+};
+
+/**
+ * Sets the clipboard data that can be pasted with
+ * `pressKeyWithModifier( 'primary', 'v' )`.
+ *
+ * @param  this
+ * @param  clipboardData
+ * @param  clipboardData.plainText
+ * @param  clipboardData.html
+ */
+export function setClipboardData(
+	this: PageUtils,
+	{ plainText = '', html = '' }: typeof clipboardDataHolder
+) {
+	clipboardDataHolder = {
+		plainText,
+		html,
+	};
+}
+
+async function emulateClipboard( page: Page, type: 'copy' | 'cut' | 'paste' ) {
+	clipboardDataHolder = await page.evaluate(
+		( [ _type, _clipboardData ] ) => {
+			const clipboardDataTransfer = new DataTransfer();
+
+			if ( _type === 'paste' ) {
```

Comment:
Logic is different here from the previous `emulateClipboard` util. Was there a bug or something else?

Reply:
Commented [here](https://github.com/WordPress/gutenberg/pull/39807/files#r857489983).